pkg/pb/synthetic_monitoring/checks_extra.go

```diff
@@ -236,6 +231,27 @@ func (p *Probe) Validate() error {
 	return nil
 }
 
+func (l Label) Validate() error {
+	if len(l.Name) == 0 {
+		return ErrInvalidLabelName
+	}
+
+	// This bit is lifted from Prometheus code, except that
+	// Prometheus accepts /^[a-zA-Z_][a-zA-Z0-9_]*$/ and we accept
+	// /^[a-zA-Z0-9_]+$/ because these names are going to be
+	// prefixed with "label_".
+	for _, b := range l.Name {
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || (b >= '0' && b <= '9')) {
```

Comment:
curious to know why we didn't use https://golang.org/pkg/regexp/ here.

Reply:
According to Prometheus code it's for performance reasons. I doubt it makes a difference here, they probably run this bit of code many times, we only do it when checks are added.